controllers/tykoasapidefinition_controller.go

```diff
@@ -175,22 +176,70 @@ func (r *TykOasApiDefinitionReconciler) createOrUpdateTykOASAPI(ctx context.Cont
 		apiID = EncodeNS(client.ObjectKeyFromObject(tykOASCrd).String())
 	}
 
+	if tykOASCrd.Spec.ClientCertificateRef != nil {
+		if tykOASDoc, err = r.processClientCertificateRef(ctx, env, log, tykOASCrd, tykOASDoc); err != nil {
+			log.Error(err, "failed to process client certificate")
```

Comment:
do we want to print this message as returning errors cause operator manager to print out the `err` as well?

Reply:
I would like to keep it as it gives context of which operation failed because of the error.